test/group_pool_wf_test.go

```diff
@@ -0,0 +1,328 @@
+			fmt.Printf("Non default Pool Name %+v\n", nonDefaulPoolName)
+			fmt.Printf("Non default Pool ID %+v\n", nonDefaulPoolID)
+		}
+	}
+	defaultPool := suite.GetPoolDetail("default")
+	defaultPoolID := *defaultPool.ID
+	fmt.Printf("Default Pool ID %+v\n", defaultPoolID)
+
+	// Pool Merge - Merge pool to 'default' pool
+	_, err := suite.poolService.MergePool(nonDefaulPoolID, defaultPoolID, nil)
+	assert.Nilf(suite.T(), err, "Pool Merge failed")
+}
+
+// Edit Pool - remove array from pool
+func (suite *GroupPoolWorkflowSuite) TestModifyPoolArray() {
+	// defaultPool, _ := suite.poolService.GetPoolByName("default")
+	defaultPool := suite.GetPoolDetail("default")
+	exisitingPoolArrayList := defaultPool.ArrayList
+	defaultPoolID := *defaultPool.ID
+	arrayCount := int(*defaultPool.ArrayCount)
+	fmt.Printf("Default Pool Array Count %+v\n", arrayCount)
+
+	// Get one arrayId to retain in pool. Others Arrays will be unassigned from pool
+	arrayID := *exisitingPoolArrayList[0].ID
+	arraylist := &nimbleos.NsArrayDetail{
+		ID:      param.NewString(arrayID),
+		ArrayId: param.NewString(arrayID),
+	}
+
+	var updateArrayList []*nimbleos.NsArrayDetail
+	updateArrayList = append(updateArrayList, arraylist)
+
+	updatePoolInput := &nimbleos.Pool{
+		ArrayList: updateArrayList,
+	}
+	_, err := suite.poolService.UpdatePool(defaultPoolID, updatePoolInput)
+	assert.Nilf(suite.T(), err, "Pool Array list update failed")
+	fmt.Printf("Updated pool array list\n")
+	time.Sleep(5 * time.Second)
+}
+
+// Check if the element is present in the provided list/array/slice
+func contains(slice []string, val string) (int, bool) {
+	for i, item := range slice {
+		if item == val {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
+// Create, Modify, Delete new pool with unassigned array
+func (suite *GroupPoolWorkflowSuite) TestNewPoolCreateModifyDelete() {
+	// Get default pool details to get assigned arrays details
+	fmt.Printf("Starting new pool creation")
+	filter := &param.GetParams{}
+	groupResp, _ := suite.arraygroupService.GetGroups(filter)
+	defaultPool := suite.GetPoolDetail("default")
+	fmt.Printf("Pool array list %+v\n", defaultPool.ArrayList)
+	groupMemberList := groupResp[0].MemberList
+	fmt.Printf("Group memeber array list %+v\n", groupMemberList)
+	arrayCount := int(*defaultPool.ArrayCount)
+	var exisitingPoolArrayList []string
+	for i := 0; i < arrayCount; i++ {
+		exisitingPoolArrayList = append(exisitingPoolArrayList, *defaultPool.ArrayList[i].Name)
+	}
+	fmt.Printf("Array(s) Assigned to default pool %+v\n", exisitingPoolArrayList)
+
+	// Get an Array name which is not assigned to default Pool
+	var unassignedArrayName string
+	fmt.Printf("initiated Unassigned array name %+v\n", unassignedArrayName)
+	for i := 0; i < len(groupMemberList); i++ {
+		_, contains := contains(exisitingPoolArrayList, *groupMemberList[i])
+		if !contains {
+			unassignedArrayName = *groupMemberList[i]
+			break
+		}
+	}
+	fmt.Printf("Unassigned array name %+v\n", unassignedArrayName)
```

Comment:
Since these tests are added with group merge tests, we will also be performing pool merge & unassign one array from the pool. So there will be a free array(unassignedArrayName) for new pool creation. So this is not a prerequisite.